File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p03625/p03625_num6_parsed.cpp
#include <iostream>
#include <vector>
#include <algorithm>

int main() {
    int N;
    std::cin >> N;
    
    std::vector<int> sticks(N);
    for (int i = 0; i < N; i++) {
        std::cin >> sticks[i];
    }
    
    std::sort(sticks.begin(), sticks.end(), std::greater<int>());
    
    int maxArea = 0;
    for (int i = 0; i < N - 3; i++) {
        if (sticks[i] == sticks[i+1] && sticks[i+2] == sticks[i+3]) {
            int area = sticks[i] * sticks[i+2];
            maxArea = std::max(maxArea, area);
        }
    }
    
    std::cout << maxArea << std::endl;
    
    return 0;
}

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p03625/p03625_num6_parsed.py
def find_max_rectangle_area(N, sticks):
    count = {}
    for stick in sticks:
        if stick not in count:
            count[stick] = 0
        count[stick] += 1

    rectangle_area = 0

    for stick in count:
        if count[stick] >= 2:
            if count[stick] >= 4:
                rectangle_area = max(rectangle_area, stick * stick)
            else:
                for other_stick in count:
                    if other_stick != stick and count[other_stick] >= 2:
                        rectangle_area = max(rectangle_area, stick * other_stick)

    return rectangle_area


# Read input from standard input
N = int(input().strip())
sticks = list(map(int, input().strip().split()))

# Find maximum possible area of rectangle
max_area = find_max_rectangle_area(N, sticks)

# Print the result
print(max_area)